service.watchedlist/resources/lib/utils.py

```diff
@@ -0,0 +1,243 @@
+    buggalo.addExtraData('setting_starttype', getSetting("starttype"));
+    buggalo.addExtraData('setting_interval', getSetting("interval"));
+    buggalo.addExtraData('setting_progressdialog', getSetting("progressdialog"));
+    buggalo.addExtraData('setting_watch_user', getSetting("watch_user"));
+    buggalo.addExtraData('setting_extdb', getSetting("extdb"));
+    buggalo.addExtraData('setting_dbpath', getSetting("dbpath"));
+    buggalo.addExtraData('setting_dbfilename', getSetting("dbfilename"));
+    buggalo.addExtraData('setting_dbbackup', getSetting("dbbackup"));
+    buggalo.addExtraData('setting_db_format', getSetting("db_format"));
+    buggalo.addExtraData('setting_mysql_server', getSetting("mysql_server"));
+    buggalo.addExtraData('setting_mysql_port', getSetting("mysql_port"));
+    buggalo.addExtraData('setting_mysql_db', getSetting("mysql_db"));
+
+    
+def translateSMB(path):
+    """"translate "smb://..." to "\\..." in windows. Don't change other paths
+    
+    Args:
+        path: Path to File in Linux notation
+        
+    Returns:
+        path: Path to File in Windows notation
+    """
+    if os.sep == '\\': # windows os
+        res_smb = re.compile('smb://(\w+)/(.+)').findall(path)
+        if len(res_smb) == 0:
+            # path is not smb://...
+            return path
+        else:
+            # create smb path
+            return '\\\\'+res_smb[0][0]+'\\'+res_smb[0][1].replace('/', '\\')
+    else:
+        # linux os. Path with smb:// is correct, but can not be accessed with normal python file access
+        return path
+    
+def fileaccessmode(path):
+    """"determine file access mode in case of smb share no direct access possible
+    
+    Args:
+        path: Path to File
+        
+    Returns:
+        copy_mode: Mode of file access: 'copy' or 'normal'
+    """
+    
+    res_smb = re.compile('smb://(\w+)/(.+)').findall(path)
+    res_nw = re.compile('(\w+)://(.*?)').findall(path)
+    if os.sep == '\\': # windows os
+        if len(res_smb) != 0:
+            # smb accessable in windows
+            return 'normal'
+        elif len(res_nw) != 0:
+            # path is ftp or nfs network
+            return 'copy'
+        else:
+            # path is not smb://...
+            return 'normal'
+    else:
+        # linux os. 
+        if len(res_nw) != 0:
+            # Path with smb:// or ftp:// is correct, but can not be accessed with normal python file access
+            # use virtual file system
+            return 'copy'
+        else:
+            # "normal" path
+            return 'normal'
+
+def sleepsafe(waittime):
+    """"sleep waittime return if a shutdown is requested
+    
+    Args:
+        waittime: Time in [seconds] to wait
+        
+    Returns:
+        0 waited the requested time
+        1 shutdown detected within waiting time. Aborted waiting
+    """
+    starttime = time.time()
+    while not xbmc.abortRequested:
```

Comment:
this is depracated, see http://kodi.wiki/view/Service_add-ons
use Monitor class, please.